packages/core/core/prompt/attachments-to-parts.ts

```diff
@@ -0,0 +1,39 @@
+import { Attachment } from '@ai-sdk/ui-utils';
+import { ImagePart, TextPart } from './content-part';
+
+type ContentPart = TextPart | ImagePart;
+
+/**
+ * Converts a data URL of type text/* to a text string.
+ */
+export function dataUrlToText(dataUrl: string): string {
+  return atob(dataUrl.split(',')[1]);
```

Comment:
Now I remember: it's deprecated in node: https://stackoverflow.com/questions/68849233/convert-a-string-to-base64-in-javascript-btoa-and-atob-are-deprecated

Reply:
There was some thing with Buffer as well re browser compat (the area is somewhat tricky). I don't think `Buffer` is available in browsers. I just searched the code and found that I'm in fact still using `atob` (sorry about the confusion): https://github.com/vercel/ai/blob/main/packages/provider-utils/src/uint8-utils.ts#L3